src/lib/templateengine.js

```diff
@@ -498,7 +499,10 @@ function TemplateEngine( undefined ) {
         downtime:        0,
         alreadyCanceled: false
       };
-    
+
+    var touchStartX = null;
```

Comment:
please don't use new `var` for each line. Just reuse the `var` above from line 488 ff.